pkg/auth/ldaputil/testclient/testclient.go

```diff
@@ -143,3 +143,23 @@ func (c *DNMappingClient) Search(searchRequest *ldap.SearchRequest) (*ldap.Searc
 
 	return c.Client.Search(searchRequest)
 }
+
+// NewPagingOnlyClient returns a new PagingOnlyClient sitting on top of the parent client. This client returns the
+// provided search response for any calls to SearchWithPaging, or defers to the parent if the call is not to the
+// paged search function.
+func NewPagingOnlyClient(parent ldap.Client, response *ldap.SearchResult) ldap.Client {
```

Comment:
These clients are made to be layered and this allows for the minimal amount of changes for insertion? I don't feel strongly.